### DISCUSSION TITLE
Need to be able to use dates in the label template

### DISCUSSION THREAD

**Discussion by slobdellreachout** [gocd/gocd#12702] [Ideas]
Calendar versioning is pretty big these days and works perfectly for my current scenario since we deploy frequently and doesn't make sense to have to manually update the label template every few days.

Is this on the radar at all?  I can't find much on it other than someone else asking about it.  Thanks

**Comment by chadlwilson** [comment]
It's not possible to be able to do that right now, but if there is a good/clean way to implement it with a standard format a PR would be considered. We could re-open the old stale ticket #3236 and link to there.

A workaround would be to version your deployable artifacts using `${COUNT}` inside the template, or just use `GO_PIPELINE_COUNTER` or other ["standard" GoCD-provided env vars](https://docs.gocd.org/current/faq/dev_use_current_revision_in_build.html#standard-gocd-environment-variables) directly in your build automation and dont worry about the label.

Most things I have worked with in GoCD that are deployed continuously use an approach like this which adds in the counter, or a count of git revisions rather than relying on the label template. If you use the counter, artifacts can be reasonably easy correlated to the individual build run, but with the downside that it's coupled to the history of that pipeline within GoCD, which has some downsides if/when you want to move to a different tool/approach.

Just to keep in mind that unfortunately GoCD does not have very active feature development right now - there's a bit more history at https://www.gocd.org/2023/02/13/gocd-project-status/ if you are interested.
